src/codeu/chat/server/Controller.java

```diff
@@ -147,6 +154,51 @@ public Conversation newConversation(Uuid id, String title, Uuid owner, Time crea
     return conversation;
   }
 
+  // Search for a user in the database and create the User object in the return.
+  @Override
+  public User searchUserInDatabase(String username, String pswd){
+    StringBuilder stringBuilder = new StringBuilder();
+    Gson gson = new Gson();
+
+    // Open the database file.
+    try (BufferedReader br = new BufferedReader(new FileReader("../src/codeu/chat/databases/users"))) {
+
+      // Append the current line into the StringBuilder.
+			String sCurrentLine;
+			while ((sCurrentLine = br.readLine()) != null) {
+				stringBuilder.append(sCurrentLine);
+			}
+
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+
+    // Parse the database json into DatabaseUser array using Gson.
+    DatabaseUser[] databaseUsers = gson.fromJson(stringBuilder.toString(), DatabaseUser[].class); 
+
+    // Iterate in every DatabaseUser
+    for(DatabaseUser databaseUser : databaseUsers){
+
+      // Check if the username and pswd matches with the DatabaseUser.
+      if(databaseUser.checkLogin(username,pswd)){
+
+        final User user = new User(databaseUser);
+
+        // Check if the user is already in the server model and return it.
+        if(model.userById().first(user.id) == null){
+          model.add(user);
+          LOG.info("User created and logged in (user.id=%s user.name=%s user.display_name=%s user.time=%s)",user.id,user.name,user.display_name,user.creation);
```

Comment:
It already has a "toString()" form but that code was implemented by google, not by me. I only used it.